### PR TITLE
implement download_fdz and download_pisa prototypes

### PR DESCRIPTION
**Pull request by beckerbenj**
For comparing OECD and FDZ data sets, download functions are requested. This is a first implementation of both.

### CODE REVIEW

**Review by weigtnik — CHANGES_REQUESTED** [review]
I think this is a great functionality! I found some relatively minor issues in the code. Apart from that, LGMT!

File: R/download_pisa.R [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  old_timeout <- getOption("timeout")
+  options(timeout = max(300, old_timeout))
+  on.exit(options(timeout = old_timeout))
+
+  ### download data to a folder with writing permissions
+  download.file(url = zip_path,
+                destfile = file.path(temp_folder, "pisa2018.zip"))
+  # use progress bar?
+  #http://r-lib.github.io/progress/
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply R/download_pisa.R:42]: Realized this very late, as the progress bar opens as a separate task in the background with my RStudio setup. Was this different for you?

File: R/download_pisa.R [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  year <- match.arg(year)
+  data_type <- match.arg(data_type)
+
+  ## choose path
+  if(year == "2018" && data_type == "stud_quest") {
+    zip_path <- "https://webfs.oecd.org/pisa2018/SPSS_STU_QQQ.zip"
+    data_subdir <- "STU/CY07_MSU_STU_QQQ.sav"
+  } else if(year == "2015" && data_type == "stud_quest") {
+    zip_path <- "https://webfs.oecd.org/pisa/PUF_SPSS_COMBINED_CMB_STU_QQQ.zip"
+    data_subdir <- "CY6_MS_CMB_STU_COG.sav"
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply R/download_pisa.R:26]: Fixed, thanks for spotting this!

File: tests/testthat/test-donwload_pisa.R [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+test_that("download pisa 2018 student questionaire data", {
+
+  skip_if_not(isTRUE(as.logical(Sys.getenv("CI", "false"))))
+
+  old_timeout <- getOption("timeout")
+  pisa2018 <- download_pisa(year = "2018", data_type = "stud_quest")
+
+  expect_equal(ncol(pisa2018), 1119)
+  expect_equal(nrow(pisa2018), 1)
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply tests/testthat/test-donwload_pisa.R:12]: Fixed, thanks for spotting this!

File: tests/testthat/test-donwload_pisa.R [hunk 4]
```diff
@@ -0,0 +1,16 @@
+
+
+
+test_that("download pisa 2018 student questionaire data", {
+
+  skip_if_not(isTRUE(as.logical(Sys.getenv("CI", "false"))))
```
[the remark this answers is not in the document]
Comment (beckerbenj) [reply tests/testthat/test-donwload_pisa.R:6]: This line ensures that this test is only executed on the Github Action machines. 

I chose this as the download of the zip-files is rather time consuming and I do not want this done every time I call `devtools::test()` locally. Of course, this has the downside that one might miss some problems with the function locally (as it just happened to me...).

File: man/download_pisa.Rd
[the remark this answers is not in the document]
Comment (beckerbenj) [reply man/download_pisa.Rd:1]: Agree and modified as suggested.

File: R/download_pisa.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (weigtnik) [reply R/download_pisa.R:43]: No, it was the same for me. I only stumbled upon it while Alt+Tab'ing to another window, too.

File: tests/testthat/test-donwload_pisa.R (hunk 4 above)
[the remark this answers is not in the document]
Comment (weigtnik) [reply tests/testthat/test-donwload_pisa.R:6]: Ah, I see. Thanks for explaining! Maybe, once the other PISA cycles are implemented too, we could test if a data set for one of the years is small enough to be loaded in a reasonable amount of time.

**Review by weigtnik — APPROVED** [review]
Thanks for implementing the requested and suggested changes! LGTM!

FYI: I ran the tests for `download_pisa` on my Windows machine (without lines 6 and 18, of course), and they were successful. 🤷

**Pull request merged**